### ISSUE TITLE
Wrong/Outdated Swift code example at 'Twitter login' documentation

### ISSUE THREAD

**Issue by idenfox** [firebase/firebase-ios-sdk#7902]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.4
  * Firebase SDK version: 7.6.0
  * Installation method: CocoaPods
  * Firebase Component: Auth

#### Steps to reproduce:

Copy and paste the step 5 from 'Handle the sign-in flow with the Firebase SDK':

https://firebase.google.com/docs/auth/ios/twitter-login

What happened? How can we make the problem occur?

The code just doesn't compile, it gives error with optional null on credential object, even when credential is != nil.

#### Relevant Code:

```
    provider.getCredentialWith(nil) { credential, error in
      if error != nil {
        // Handle error.
      }
      if credential != nil {
        Auth().signIn(with: credential) { authResult, error in
          if error != nil {
            // Handle error.
          }
          // User is signed in.
          // IdP data available in authResult.additionalUserInfo.profile.
          // Twitter OAuth access token can also be retrieved by:
          // authResult.credential.accessToken
          // Twitter OAuth ID token can be retrieved by calling:
          // authResult.credential.idToken
          // Twitter OAuth secret can be retrieved by calling:
          // authResult.credential.secret
        }
      }
    }
    
```

FIX: The solution I met was to replace 'Auth()' with 'Auth.auth'
I think documentation is outdated, it take me a day for fix because I'm new with swift. Code fixed:

    provider.getCredentialWith(nil) { credential, error in
      if error != nil {
        // Handle error.
      }
      if credential != nil {
        Auth.auth().signIn(with: credential) { authResult, error in
          if error != nil {
            // Handle error.
          }
          // User is signed in.
          // IdP data available in authResult.additionalUserInfo.profile.
          // Twitter OAuth access token can also be retrieved by:
          // authResult.credential.accessToken
          // Twitter OAuth ID token can be retrieved by calling:
          // authResult.credential.idToken
          // Twitter OAuth secret can be retrieved by calling:
          // authResult.credential.secret
        }
      }
    }

**Comment by rizafran** [comment]
Hi @idenfox,

Thanks for reporting and sharing your findings. Per checking, I see that the [Auth quickstart](https://github.com/firebase/quickstart-ios/tree/master/authentication) is currently using the `Auth.auth().signIn(with: credential)`, but the doc seems outdated as it is still using the `Auth().signIn(with: credential)`.

**Comment by morganchen12** [comment]
Docs fixed in cl/368711142 (Googlers-only link).
